content/en/blog/falcosidekick-reponse-engine-part-4-tekton.md

```diff
@@ -0,0 +1,612 @@
+Sun May  2 18:00:10 2021: Loading rules from file /etc/falco/falco_rules.local.yaml:
+Sun May  2 18:00:10 2021: Loading rules from file /etc/falco/rules.d/rules_user_known_k8s_api_callers.yaml:
+Sun May  2 18:00:10 2021: Starting internal webserver, listening on port 8765
+{"output":"20:24:10.361728219: Notice A shell was spawned in a container with an attached terminal (user=root user_loginuid=-1 k8s.ns=falcoresponse k8s.pod=alpine container=6ac7d190134e shell=sh parent=runc cmdline=sh -c uptime terminal=34816 container_id=6ac7d190134e image=alpine) k8s.ns=falcoresponse k8s.pod=alpine container=6ac7d190134e k8s.ns=falcoresponse k8s.pod=alpine container=6ac7d190134e","priority":"Notice","rule":"Terminal shell in container","time":"2021-05-02T20:24:10.361728219Z", "output_fields": {"container.id":"6ac7d190134e","container.image.repository":"alpine","evt.time":1619987050361728219,"k8s.ns.name":"falcoresponse","k8s.pod.name":"alpine","proc.cmdline":"sh -c uptime","proc.name":"sh","proc.pname":"runc","proc.tty":34816,"user.loginuid":-1,"user.name":"root"}}
+```
+
+In **Terminal 2** you should see a pod starting and hopefully Complete without any errors and the alpine pod getting killed.
+
+```shell
+NAME                                                              READY   STATUS            RESTARTS   AGE
+alpine                                                            0/1     Terminating       0          1m7s
+el-falco-listener-557786f598-znzk9                                1/1     Running           0          10m
+falco-pod-delete-pipeline-run-w2vf8-run-pod-delete-jlxl7--mk44k   0/1     Completed         0          59s
+
+```
+
+> Hurray our "hacked" pod have been killed
+
+If you look in the logs of the task
+
+```shell
+kubectl logs -f $(kubectl get pods -l tekton.dev/task=pod-delete -o jsonpath="{.items[0].metadata.name}" -n falcoresponse) -n falcoresponse
+2021/05/02 18:11:00 PodName: alpine & Namespace: falcoresponse
+2021/05/02 18:11:00 Rule: Terminal shell in container
+2021/05/02 18:11:00 Deleting pod alpine from namespace falcoresponse
+```
+
+## Conclusion
+
+This was a rather simple example on how we can use the power of tekton together with Falco to protect us from bad actors that is trying to take over pods in our cluster.
+
+As noted during this post there are allot of potential improvements before this is production ready:
+
+- The criticalNamepsaces in our go code is currently hard-coded and needs to be input variable of some kind.
+- We need to be able to delete pods depending on priority level, rule or something similar.
+- To be able to debug pods we might need to shell in to them, we need a way to ignore pods temporary without the pod getting restarted. Probably a annotation to look for in the pod before deleting it.
+- And probably many other needs that you can come up with.
+
+If you have any ideas/issues come and share them in the falco slack [https://kubernetes.slack.com](https://kubernetes.slack.com) #falco.
```

Comment:
add this 

Falco
If you would like to find out more about Falco:

Get started in Falco.org.
Check out the Falco project in GitHub
Get involved in the Falco community.
Meet the maintainers on the Falco Slack.
Follow @falco_org on Twitter.

as the other blogs in this series https://falco.org/blog/falcosidekick-reponse-engine-part-3-knative/

perhaps @NissesSenap / @Issif   we should also put details on Tekton (for more information?)

Reply:
Done, also linked back to tekton